packages/bsky/src/api/com/atproto/label/queryLabels.ts

```diff
@@ -0,0 +1,33 @@
+import { Server } from '../../../../lexicon'
+import AppContext from '../../../../context'
+import { InvalidRequestError } from '@atproto/xrpc-server'
+
+export default function (server: Server, ctx: AppContext) {
+  server.com.atproto.label.queryLabels(async ({ params, req }) => {
+    const { uriPatterns, sources } = params
+
+    if (uriPatterns.find((uri) => uri.includes('*'))) {
+      throw new InvalidRequestError('wildcards not supported')
+    }
+
+    const labelMap = await ctx.hydrator.label.getLabelsForSubjects(
+      uriPatterns,
+      // If sources are provided, use them. Otherwise, use the labelers from the request header
+      sources
```

Comment:
Not sure if this creates some kind of confusion because someone might pass different labelers through header and then pass different labelers through the source param and get results only for the source param.

Reply:
We can always ignore the header values here.

Reply:
I guess my question is, should we? to me, it feels a bit nicer to be able to just send it via header and ignore source param entirely.

Reply:
In my view it's not really the purpose of the header, which is about looking at the content of the response and applying labels to it.  This endpoint doesn't really have serve any contents that can be labeled, so it's a little bit outside the purview of the header.  As you pointed out, it may also cause confusion because there's a parameter dedicated to querying specific labelers, and it's not well-defined how that parameter and the header should interact with each other.  By relying only on the dedicated parameter here, I think it makes the behavior pretty clear/predictable.  @bnewbold @dholms any thoughts to add?

Reply:
alright we're no longer defaulting to labelers in header and throwing error if sources param is missing.